MekHQ/src/mekhq/MHQStaticDirectoryManager.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2020 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq;
+
+import megamek.MegaMek;
+import megamek.client.ui.swing.tileset.MMStaticDirectoryManager;
+import megamek.common.Crew;
+import megamek.common.annotations.Nullable;
+import megamek.common.util.fileUtils.DirectoryItems;
+import mekhq.campaign.force.Force;
+import mekhq.gui.enums.LayeredForceIcon;
+import mekhq.gui.utilities.PortraitFileFactory;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Vector;
+
+public class MHQStaticDirectoryManager extends MMStaticDirectoryManager {
+    //region Variable Declarations
+    private static DirectoryItems forceIconDirectory;
+    private static DirectoryItems awardIconDirectory;
+
+    // Re-parsing Prevention Variables: The are True at startup and when the specified directory
+    // should be re-parsed, and are used to avoid re-parsing the directory repeatedly when there's an error.
+    private static boolean parseForceIconDirectory = true;
+    private static boolean parseAwardIconDirectory = true;
+    //endregion Variable Declarations
+
+    //region Constructors
+    protected MHQStaticDirectoryManager() {
+        // This class is not to be instantiated
+    }
+    //endregion Constructors
+
+
+    //region Initialization
+    /**
+     * This initialized all of the directories under this manager
+     */
+    public static void initialize() {
+        MMStaticDirectoryManager.initialize();
+        initializeForceIcons();
+        initializeAwardIcons();
+    }
+
+    /**
+     * Parses MekHQ's force icon folder when first called or when it was refreshed.
+     *
+     * @see #refreshForceIcons()
+     */
+    private static void initializeForceIcons() {
```

Comment:
MekHQ is certainly multi-threaded compared to MM, I feel that these individual initialize methods should be synchronized.